src/commands/serverboard.ts

```diff
@@ -0,0 +1,101 @@
+import {
+  SlashCommandSubcommandBuilder, ButtonBuilder, ActionRowBuilder,
+  ButtonStyle, ButtonInteraction, type ChatInputCommandInteraction
+} from "discord.js";
+import { quickSort } from "../utils/quickSort.js";
+import { serverEmbed } from "../utils/embeds/serverEmbed.js";
+import { database, getNewsTable, getServerboardTable } from "../utils/database.js";
+import { QuickDB } from "quick.db";
+
+export default class Serverboard {
+  data: SlashCommandSubcommandBuilder;
+  db: QuickDB<any>;
+
+  constructor(db?: QuickDB) {
+    this.db = db;
+    this.data = new SlashCommandSubcommandBuilder()
+      .setName("serverboard")
+      .setDescription("Shows the servers that have Nebula.")
+      .addNumberOption(option => option
+        .setName("page")
+        .setDescription("The page you want to see.")
+      );
+  }
+
+  async run(interaction: ChatInputCommandInteraction) {
+    const db = this.db;
+    const guilds = interaction.client.guilds.cache;
+    const guildsMapped = {};
+    const shownGuilds = (await (await getServerboardTable(db)).all().catch(() => [])) as any[];
+
+    for (const guild of guilds.values()) {
+      const shownVal = shownGuilds?.find(shown => shown?.id == guild.id)?.value?.shown;
+      const isShown = shownVal == null ? null : shownVal;
+
+      if (isShown == false) continue;
+      if (isShown == null && guild?.rulesChannelId != null) continue;
+
+      guildsMapped[guild.memberCount + ":" + guild.id] = guild;
+    }
+
+    const sortedGuilds = quickSort(
+      [...Object.keys(guildsMapped).map(i => Number(i.split(":")[0]))],
+      [[...Object.values(guildsMapped)]],
+      0,
+      Object.keys(guildsMapped).length - 1
+    );
+
+    const guildsSorted = sortedGuilds[1][0].reverse();
+    const pages = guildsSorted.length;
+    const argPage = interaction.options.getNumber("page", false);
+    let page = (argPage - 1 <= 0 ? 0 : argPage - 1 > pages ? pages - 1 : argPage - 1) || 0;
+
+    let guild = guildsSorted[page];
+    let subs = await (await getNewsTable(db))
+      ?.get(`${guild.id}.subscriptions`)
```

Comment:
```suggestion
    let subs = await (await getNewsTable(db))?
      .get(`${guild.id}.subscriptions`)
```
That character belongs there I think ;)